packages/service-generator-csharp/src/service.ts

```diff
@@ -0,0 +1,831 @@
+import {
+  BooleanLiteral,
+  EmitContext,
+  Enum,
+  IntrinsicType,
+  Model,
+  ModelProperty,
+  Namespace,
+  NumericLiteral,
+  Operation,
+  Program,
+  Scalar,
+  StringLiteral,
+  Tuple,
+  Type,
+  Union,
+  getDoc,
+  getNamespaceFullName,
+  getService,
+  isErrorModel,
+} from "@typespec/compiler";
+import {
+  CodeTypeEmitter,
+  Context,
+  Declaration,
+  EmitEntity,
+  EmittedSourceFile,
+  EmitterOutput,
+  Scope,
+  SourceFile,
+  StringBuilder,
+  TypeSpecDeclaration,
+  code,
+} from "@typespec/compiler/emitter-framework";
+import {
+  HttpOperation,
+  HttpOperationParameter,
+  HttpOperationResponse,
+  MetadataInfo,
+  Visibility,
+  createMetadataInfo,
+  getHttpOperation,
+} from "@typespec/http";
+import { getResourceOperation } from "@typespec/rest";
+import { exec } from "child_process";
+import {
+  emitOperationState,
+  emitOperationStatus,
+  emitResourceOperationStatus,
+} from "./boilerplate.js";
+import { CSharpSourceType, CSharpType, ControllerContext, NameCasingType } from "./interfaces.js";
+import { CSharpServiceEmitterOptions } from "./lib.js";
+import {
+  ensureCSharpIdentifier,
+  ensureCleanDirectory,
+  formatComment,
+  getCSharpIdentifier,
+  getCSharpStatusCode,
+  getCSharpTypeForScalar,
+  getOperationVerbDecorator,
+  hasNonMetadataProperties,
+  isValueType,
+} from "./utils.js";
+
+export async function $onEmit(context: EmitContext<CSharpServiceEmitterOptions>) {
+  let _unionCounter: number = 0;
+  const controllers = new Map<string, ControllerContext>();
+  const NoResourceContext: string = "RPCOperations";
+
+  class CSharpCodeEmitter extends CodeTypeEmitter {
+    #names: string[] = [];
+    #licenseHeader = `// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
(Curious) I understand the direct boilerplate types and the generator code itself falling under the MIT license and the Microsoft copyright.

But does this license header end up on “my” generated files from my TypeSpec? I would expect that the types generated would NOT include this license header.

Reply:
This is a good point, we should make this configurable so that generated code could fall under any selected license (or have no license header).